File: InsuraTech.Domain/Common/PolicyStatusHistory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using InsuraTech.Domain.Policies;

namespace InsuraTech.Domain.Common
{
    public sealed class PolicyStatusHistory : Entity
    {
        public Guid PolicyId { get; private set; }
        public PolicyStatus Status { get; private set; }
        public string Notes { get; private set; } = null!;
        public DateTime ChangedAt { get; private set; }

        private PolicyStatusHistory() { }

        public static PolicyStatusHistory Create(Guid policyId, PolicyStatus status, string notes)
        {
            return new PolicyStatusHistory
            {
                PolicyId = policyId,
                Status = status,
                Notes = notes,
                ChangedAt = DateTime.UtcNow
            };
        }
    }
}

File: backend_v2/src/main/java/com/insuratech/domain/common/PolicyStatusHistory.java
package com.insuratech.domain.common;

import lombok.Getter;

import java.time.Instant;

@Getter
public class PolicyStatusHistory {

    private final String status;
    private final Instant changedAt;
    private final String changedBy;
    private final String reason;

    public PolicyStatusHistory(String status, String changedBy, String reason) {
        this.status = status;
        this.changedAt = Instant.now();
        this.changedBy = changedBy;
        this.reason = reason;
    }
}
